t04_refactored.py

```diff
@@ -267,8 +267,69 @@ def team_21_adv(username, delay):
 
 
 def team_22_adv(username, delay):
-    pass
-    # TODO Add your code here
+    """
+    https://docs.google.com/document/d/1KfNXomNLKmDJjPjTKwPYSAfdsDXUeLUHza7j-Q3XNqo/edit
+    Harry Adkins
+    :return: none
+    """
+    talk = input("Do you wish to talk to them? [Yes/No]")
+
+    if talk == "Yes":
+        print("It seems they're just as lost as you are!")
+        sleep(delay)
+        print("You agree to help each other find your way out of this cave, and they have tools to help")
+    else:
+        print("Due to your silence and shadowy figure in the dark, they mistake you for a beast!")
+        chance = input("How many steps do you take towards the person? Enter an integer")
+        if int(chance) > 7:
+            print("They are startled by your rapid approach!")
+            print("They charge at you head on and you trip and fall.")
+            sleep(delay * 2)
+            print("You keep falling?")
+            sleep(delay)
+            print("It seems you fell over a cliff, you're still falling and you don't know when you'll hit the gro-")
+            dead = True
+        else:
+            print("They get closer to you and see you're a human too!")
+            print("They agree to help you using tools from their backpack")
+
+    if dead == True:
+        print("Oh no! You died. Better luck next time! Try again by hitting the green play button. ")
+        quit()
```

Comment:
Instead of killing directly, `return False` will end the function, and the next line in `main()` is `kill_if_dead()`, which will do the same thing.